crates/net/network/src/transactions.rs

```diff
@@ -276,67 +276,114 @@ where
         let max_num_full = (self.peers.len() as f64).sqrt() as usize + 1;
 
         // Note: Assuming ~random~ order due to random state of the peers map hasher
-        for (peer_idx, (peer_id, peer)) in self.peers.iter_mut().enumerate() {
-            // filter all transactions unknown to the peer
-            let mut hashes = PooledTransactionsHashesBuilder::new(peer.version);
-            let mut full_transactions = FullTransactionsBuilder::default();
-
-            // Iterate through the transactions to propagate and fill the hashes and full
-            // transaction lists, before deciding whether or not to send full transactions to the
-            // peer.
-            for tx in to_propagate.iter() {
-                if peer.transactions.insert(tx.hash()) {
-                    hashes.push(tx);
-
-                    // Do not send full 4844 transaction hashes to peers.
-                    //
-                    //  Nodes MUST NOT automatically broadcast blob transactions to their peers.
-                    //  Instead, those transactions are only announced using
-                    //  `NewPooledTransactionHashes` messages, and can then be manually requested
-                    //  via `GetPooledTransactions`.
-                    //
-                    // From: <https://eips.ethereum.org/EIPS/eip-4844#networking>
-                    if !tx.transaction.is_eip4844() {
-                        full_transactions.push(tx);
-                    }
-                }
-            }
-            let mut new_pooled_hashes = hashes.build();
+        for (peer_idx, peer_info) in self.peers.iter_mut().enumerate() {
+            Self::perform_peer_propagation(
```

Comment:
not a fan of this change, I want to keep the existing function untouched

Reply:
Could you give some motivation for why? If the logic is going to be exactly the same, then why not reduce the loc? Unless I misunderstood the purpose of the function

Reply:
because it's not exactly same, and I don't like this change, I want to keep these things separate, the manual command should not mess with the actual network msg handling.

Reply:
Reverted the change

> the manual command should not mess with the actual network msg handling.

Can you specify what you mean by this?